cmd/kubectl-plugin/main.go

```diff
@@ -35,44 +34,40 @@ func main() {
 		clusterName = val
 	}
 
-	kubeConfigBytes, err := exec.Command(cmdBin, "config", "view").Output()
-	if err != nil {
-		log.Fatal(err)
+	kubeconfigFilePath := getKubeConfigDefaultPath(getHomePath())
+	if len(kubeconfigFilePath) == 0 {
+		log.Fatal("error initializing config. The KUBECONFIG environment variable must be defined.")
 	}
 
-	tmpConf, err := ioutil.TempFile("", "example")
+	config, err := configFromPath(kubeconfigFilePath)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("error obtaining kubectl config: %v", err)
 	}
 
-	defer os.Remove(tmpConf.Name()) // clean up
-	if _, err = tmpConf.Write(kubeConfigBytes); err != nil {
-		log.Fatal(err)
-	}
-	// use the current context in kubeconfig
-	kubeConfig, err := clientcmd.BuildConfigFromFlags("", tmpConf.Name())
+	rest, err := config.ClientConfig()
 	if err != nil {
-		panic(err.Error())
+		log.Fatalf(err.Error())
 	}
 
-	redisClient, err := rclient.NewClient(kubeConfig)
+	redisClient, err := rclient.NewClient(rest)
 	if err != nil {
 		glog.Fatalf("Unable to init redis.clientset from kubeconfig:%v", err)
 	}
 
-	var rcs *v1.RedisClusterList
+	rcs := &v1.RedisClusterList{}
 	if clusterName == "" {
 		rcs, err = redisClient.RedisoperatorV1().RedisClusters(namespace).List(meta_v1.ListOptions{})
 		if err != nil {
-			glog.Fatalf("unable to list rediscluster err:%v", err)
+			glog.Fatalf("unable to list rediscluster:%s err:%v", err)
```

Comment:
miss rediscluter name?

Reply:
or namespace... not sure

Reply:
thanks for your comments: new PR#16 to fix it all